calicoctl/commands/ipam.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/tigera/libcalico-go/lib/client"
+	caliconet "github.com/tigera/libcalico-go/lib/net"
+
+	docopt "github.com/docopt/docopt-go"
+)
+
+// IPAM takes keyword with an IP address then calls the subcommands
+func IPAM(args []string) error {
+	doc := `Usage: 
+    calicoctl ipam release --ip=<IP>
+    calicoctl ipam show --ip=<IP>
+
+Description:
+    IP address management
+
+Options:
+    --ip=<IP>     The directory for logs [default: /var/log/calico] 
+
+Warnings:
+  -  Releasing an in-use IP address can result in it being assigned to multiple
+     workloads.
+`
+
+	parsedArgs, err := docopt.Parse(doc, args, true, "calicoctl", false, false)
+	if err != nil {
+		return err
+	}
+
+	// Length of parsedArgs is 0 when `-h` `--help` is passed
+	// docopts takes care of printing the help message, but we need to
+	// stop return nil here so command doesn't go any further when `-h` is passed
+	if len(parsedArgs) == 0 {
+		return nil
+	}
+
+	// create a new backend client from env vars
```

Comment:
I believe that link says to put periods at the end of both declarative, and imperative sentences, and this would be an instance of an imperative sentence.  See: http://www.mit.edu/course/21/21.guide/imperati.htm

```
Use periods to indicate the end of declarative or imperative sentences.

The Newcomen engine utilizes the condensation of steam to create a partial vacuum beneath a piston.
Expand the system's bandwidth.
```

:)